src/components/shared/chart/Canvas.js

```diff
@@ -109,28 +109,41 @@ export default class ChartCanvas<HoveredItem> extends React.Component<
     const canvas = this._canvas;
     const { containerWidth, containerHeight, scaleCtxToCssPixels } = this.props;
     const { devicePixelRatio } = window;
-    const devicePixelWidth: DevicePixels = containerWidth * devicePixelRatio;
-    const devicePixelHeight: DevicePixels = containerHeight * devicePixelRatio;
+
     if (!canvas) {
       return;
     }
-    // Satisfy the null check for Flow.
-    const ctx = this._ctx || canvas.getContext('2d', { alpha: false });
-    if (!this._ctx) {
+
+    let ctx = this._ctx;
+    if (!ctx) {
+      ctx = canvas.getContext('2d', { alpha: false });
       this._ctx = ctx;
     }
+
+    // Round down to integer values.
+    const devicePixelWidth: DevicePixels = Math.floor(
+      containerWidth * devicePixelRatio
+    );
+    const devicePixelHeight: DevicePixels = Math.floor(
+      containerHeight * devicePixelRatio
+    );
+
     if (
       canvas.width !== devicePixelWidth ||
       canvas.height !== devicePixelHeight
     ) {
+      // The canvas needs to be sized to the container.
       canvas.width = devicePixelWidth;
       canvas.height = devicePixelHeight;
-      canvas.style.width = containerWidth + 'px';
-      canvas.style.height = containerHeight + 'px';
+      // Round the style width as well, so the canvas will be sized according to
+      // the integer pixel size.
+      canvas.style.width = Math.floor(containerWidth) + 'px';
+      canvas.style.height = Math.floor(containerHeight) + 'px';
```

Comment:
I'm not sure we should round down these values here, because we used them to compute devicePixelHeight/Width before. There's risk to have some computation errors.
For example if `containerWidth` is 3.9, `devicePixelRatio` is 2, we get `devicePixelWidth == 7` (`Math.floor(3.9*2)`), but `canvas.style.width == 3`

 At least they should be rounded down _before_ we calculate devicePixelHeight/Width, or just not round them down at all.

Reply:
As discussed on Slack, I filed a bug for this as it requires more work than I have time for right now.

#2370